src/Documenter.jl

```diff
@@ -637,55 +562,54 @@ julia> Documenter.generate("MyPackageName")
 [ ... output ... ]
 ```
 """
-function generate(pkgname::AbstractString; dir=nothing)
-    # TODO:
-    #   - set up deployment to `gh-pages`
-    #   - fetch url and username automatically (e.g from git remote.origin.url)
-
-    # Check the validity of the package name
-    if length(pkgname) == 0
-        error("Package name can not be an empty string.")
-    end
-    # Determine the root directory where we wish to generate the docs and
-    # check that it is a valid directory.
-    docroot = if dir === nothing
-        pkgdir = Pkg.dir(pkgname)
-        if !isdir(pkgdir)
-            error("Unable to find package $(pkgname).jl at $(pkgdir).")
-        end
-        joinpath(pkgdir, "docs")
-    else
-        dir
-    end
+function generate(pkgname::AbstractString; dir=nothing,
+                  remote = "origin", online = true,
+                  extras = ["C:/Program Files/Git/usr/bin"] )
+
+    pkgdir = Utilities.backup_dir(dir, pkgname)
+    info("with base files")
+    docroot = joinpath(pkgdir, "docs") |> Utilities.bad_dir
+    info("Will generate documentation here")
```

Comment:
I'd say that `bad_dir` shouldn't print anything and the `info` line here should print the `docroot` instead of "here".